sdk/plugin/pb/backend.proto

```diff
@@ -453,6 +459,9 @@ service Backend {
 
    	// Type returns the BackendType for the particular backend
 	rpc Type(Empty) returns (TypeReply);
+
+	// Version returns version information for the plugin.
+	rpc Version(Empty) returns (VersionReply);
```

Comment:
Adding the method here has a couple of (minor) drawbacks:

1. We can't use a single protobuf definition for both db and auth/secret engines.
2. The original intention was that self-reporting the plugin's version would be opt-in. Anyone using this version of the SDK will have to implement the method to fulfil the `Backend` interface. I suppose they could return an empty version though.

I haven't thought too much about the benefits of the way it is now though - what's the motivation to add it here?

Reply:
SGTM, although the points above are definitely minor if you think there's good opportunity to simplify by adding to the existing interfaces.